src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,38 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+            return actor;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Failed to save actor: " + actor);
```

Comment:
It's a good practice to include the original exception when throwing a new one. This can help to debug the cause of the original exception. You can do this by providing the original exception as a parameter to the constructor of the new exception. Like this: `throw new DataProcessingException("Failed to save actor: " + actor, e);`